plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -0,0 +1,520 @@
+            String location,
+            Map<String, String> properties)
+    {
+        TableMetadata metadata = newTableMetadata(schema, partitionSpec, location, properties);
+        TableOperations tableOperations = tableOperationsProvider.createTableOperations(
+                this,
+                session,
+                schemaTableName.getSchemaName(),
+                schemaTableName.getTableName(),
+                Optional.of(session.getUser()),
+                Optional.of(location));
+        return createTableTransaction(schemaTableName.toString(), tableOperations, metadata);
+    }
+
+    @Override
+    public void renameTable(ConnectorSession session, SchemaTableName from, SchemaTableName to)
+    {
+        boolean newTableCreated = false;
+        boolean oldTableDeleted = false;
+        try {
+            GetTableRequest getTableRequest = new GetTableRequest()
+                    .withDatabaseName(from.getSchemaName())
+                    .withName(from.getTableName());
+            catalogId.ifPresent(getTableRequest::setCatalogId);
+            com.amazonaws.services.glue.model.Table table = stats.getGetTable().call(() -> glueClient.getTable(getTableRequest).getTable());
+            TableInput tableInput = new TableInput()
+                    .withName(to.getTableName())
+                    .withTableType(table.getTableType())
+                    .withOwner(table.getOwner())
+                    .withParameters(table.getParameters());
+
+            CreateTableRequest createTableRequest = new CreateTableRequest()
+                    .withDatabaseName(to.getSchemaName())
+                    .withTableInput(tableInput);
+            catalogId.ifPresent(createTableRequest::setCatalogId);
+            stats.getCreateTable().call(() -> glueClient.createTable(createTableRequest));
+            newTableCreated = true;
+
+            deleteTable(from.getSchemaName(), from.getTableName());
+            oldTableDeleted = true;
+        }
+        catch (Exception e) {
+            if (newTableCreated && !oldTableDeleted) {
+                try {
+                    deleteTable(to.getSchemaName(), to.getTableName());
+                }
+                catch (Exception cleanupException) {
+                    if (!cleanupException.equals(e)) {
+                        e.addSuppressed(cleanupException);
+                    }
+                }
+            }
+            throw e;
+        }
+    }
+
+    private void deleteTable(String schema, String table)
+    {
+        DeleteTableRequest deleteNewTableRequest = new DeleteTableRequest()
+                .withDatabaseName(schema)
+                .withName(table);
+        catalogId.ifPresent(deleteNewTableRequest::setCatalogId);
+        stats.getDropTable().call(() -> glueClient.deleteTable(deleteNewTableRequest));
+    }
+
+    @Override
+    public void updateTableComment(ConnectorSession session, SchemaTableName schemaTableName, Optional<String> comment)
```

Comment:
same as in `io.trino.plugin.iceberg.catalog.hms.TrinoHiveCatalog#updateTableComment`
it's not metastore-specific.